File: arduinobot_cpp_examples/src/simple_service_server.cpp
#include "rclcpp/rclcpp.hpp"
#include <memory>
#include "arduinobot_msgs/srv/add_two_ints.hpp"

using std::placeholders::_1;
using std::placeholders::_2;

class SimpleServiceServer : public rclcpp::Node
{
public:
    SimpleServiceServer() : Node("simple_service_server")
    {
        service_ = create_service<arduinobot_msgs::srv::AddTwoInts>(
            "add_two_ints",
            std::bind(&SimpleServiceServer::serviceCallback, this, _1, _2));
            RCLCPP_INFO(this->get_logger(), "Service server has been started...");
    }

private:
    rclcpp::Service<arduinobot_msgs::srv::AddTwoInts>::SharedPtr service_;

    void serviceCallback(const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Request> request,
                         const std::shared_ptr<arduinobot_msgs::srv::AddTwoInts::Response> response)
    {   
        RCLCPP_INFO(this->get_logger(), "New Request Received a: %ld, b: %ld", request->a, request->b);
        
        response->sum = request->a + request->b;

        RCLCPP_INFO(this->get_logger(), "Returning sum: %ld", response->sum);
    }

};

int main(int argc, char **argv)
{

    rclcpp::init(argc, argv);
    auto node = std::make_shared<SimpleServiceServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: arduinobot_py_examples/arduinobot_py_examples/simple_service_server.py
import rclpy
from rclpy.node import Node
from arduinobot_msgs.srv import AddTwoInts

class SimpleServiceServer(Node):
    def __init__(self):
        super().__init__("simple_service_server")

        self.service_ = self.create_service(AddTwoInts, "add_two_ints", self.serviceCallback)
        self.get_logger().info("Service add_two_ints is ready!")

    def serviceCallback(self, req, res):
        self.get_logger().info(f'New Message Received a:{req.a}, b:{req.b}')
        res.sum = req.a + req.b
        self.get_logger().info(f'Returning sum: {res.sum}')
        return res


def main():
    rclpy.init()
    simple_service_server = SimpleServiceServer()
    rclpy.spin(simple_service_server)
    simple_service_server.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()
